File: NYTViewer-project/app/src/main/java/com/gmail/dp/denzay/nytviewer/di/AppComponent.java
package com.gmail.dp.denzay.nytviewer.di;

import com.gmail.dp.denzay.nytviewer.AsyncDBImageDownloader;
import com.gmail.dp.denzay.nytviewer.MainActivity;
import com.gmail.dp.denzay.nytviewer.adapters.FavouritesDBAdapter;
import com.gmail.dp.denzay.nytviewer.data.DBProvider;
import com.gmail.dp.denzay.nytviewer.views.FavouritesFragment;
import com.gmail.dp.denzay.nytviewer.views.WebViewActivity;

import javax.inject.Singleton;

import dagger.Component;

@Component(modules = {AppModule.class, CacheStorageModule.class,
                      DBProviderModule.class, FavouritesDBAdapterModule.class, FavouriteCachedDBHelperModule.class, })
@Singleton
public interface AppComponent {
    void inject(MainActivity aMainActivity);
    void inject(WebViewActivity aWebViewActivity);
    void inject(FavouritesFragment aFavouritesFragment);

    void inject(FavouritesDBAdapter aFavouritesDBAdapter);
    void inject(AsyncDBImageDownloader aAsyncDBImageDownloader);
    void inject(DBProvider aDBProvider);
}

File: NYTViewerMVVM/app/src/main/java/com/gmail/dp/denzay/nytviewer_mvvm/di/components/AppComponent.kt
package com.gmail.dp.denzay.nytviewer_mvvm.di.components

import android.app.Application
import com.gmail.dp.denzay.nytviewer_mvvm.NYTViewerApp
import com.gmail.dp.denzay.nytviewer_mvvm.di.modules.AppModule
import com.gmail.dp.denzay.nytviewer_mvvm.di.modules.MainActivityModule
import dagger.BindsInstance
import dagger.Component
import dagger.android.AndroidInjectionModule
import javax.inject.Singleton

@Singleton
@Component(modules = [AndroidInjectionModule::class, AppModule::class, MainActivityModule::class])
interface AppComponent {

    @Component.Builder
    interface Builder {
        @BindsInstance
        fun application(application: Application): Builder

        fun build(): AppComponent
    }

    fun inject(app: NYTViewerApp)
}